src/components/periodic-report/periodic-report.resolver.ts

```diff
@@ -22,4 +31,25 @@ export class PeriodicReportResolver {
     );
     return reportFile;
   }
+
+  @ResolveField(() => SecuredFile, {
+    nullable: true,
+  })
+  async reportFile(
+    @Parent() report: IPeriodicReport,
+    @AnonSession() session: Session
+  ): Promise<SecuredFile | null> {
+    const { value: id, ...rest } = report.reportFile;
+    const value = id
+      ? await this.files.resolveDefinedFile(
+          {
+            value: id,
+            ...rest,
+          },
+          session
+        )
+      : null;
+
+    return value;
```

Comment:
`resolveDefinedFile` already handles all that.
```suggestion
  ): Promise<SecuredFile> {
  return await this.files.resolveDefinedFile(report.reportFile);
```